### ISSUE TITLE
Emulator crashes and save file corruption after using save states once. 

### ISSUE THREAD

**Issue by AbdullahTrees** [Arisotura/melonDS#1312]
My Pokemon Diamond run was going smoothly the past few days, until I decided to catch a Gastly with one poke ball without damaging it (by getting lucky with save states). 

As soon as I deleted the save states and tried saving the game(in game), the save happened immediately, which is a huge warning sign(because usually it takes 15-20 seconds).  So I quickly took a save state, and a few seconds later the emulator shut down without notice (most likely a crash). 

Upon reloading, I see that my save file has been erased, so if it were not for the save state I made, I'd have lost all progress!!!!

So then I tried opening my save state and saving the game from inside the menu. I seem to have created some sort of an unstable behavior where sometimes the emulator is crashing and sometimes it is not. I really don't want to have to rely on save states and wanna go back to in-game saving again ;(

Anyways, wanted to let you guys know about this unstable behavior/crash and if anyone else has reported it earlier
Attached the offending save states and save files(can't share the rom obviously, but I labelled the rom so it should be easy to find), lemme know if you guys need anything else from me. 

Thanks in advance

[1015 - Pokemon - Diamond Version (USA) (Rev 5).zip](https://github.com/Arisotura/melonDS/files/7775438/1015.-.Pokemon.-.Diamond.Version.USA.Rev.5.zip)
.

**Comment by Bonboon229** [comment]
Can attest to this,
This same problem cost me my Pokemon White 2 Save.

**Comment by AbdullahTrees** [comment]
> Can attest to this, This same problem cost me my Pokemon White 2 Save.

Thank you for sharing your experience. If we get this more attention, then it might get noticed by the devs.

**Comment by lugia19** [comment]
Yeah, this issue has been reported for a while, no fix yet. Sucks.

**Comment by RSDuck** [comment]
Which version do you use?

**Comment by Avenger3283** [comment]
I use version 0.9.5. I was playing then decided to delete my save states after i did it reverted the save file back to the save of the only remaining save state

**Comment by RSDuck** [comment]
do you have the separate savefiles option enabled?

**Comment by Avenger3283** [comment]
No I don't

**Comment by RSDuck** [comment]
I still don't really understand the issue. So you deleted the file outside of melonDS? Can you give me precise steps to reproduce the issue?

**Comment by Avenger3283** [comment]
I deleted save states from 2-9 forgot to also delete savestate 1. When I opened the game the next day the save file was back to the save I made a few minutes before I made save state 1. I tried this it happened three straight times after which I just copied my save file then deleted ml1 then never used save states again and it worked correctly again
